File: 0002.py
def fibgen(n):
    n0 = n1 = 1
    while n1 <= n:
        yield n1
        tmp = n1
        n1 = n0 + n1
        n0 = tmp

def fn0():
    return sum(i for i in fibgen(4000000) if i % 2 == 0)

solutions = [fn0]
        

File: 0002.java
public class _0002 {
    public Object fn0() {
        int f1 = 1;
        int f2 = 2;
        int sum = 0;
        while (f2 < 4000000) {
            if (f2 % 2 == 0)
                sum += f2;
            f2 = f1^f2;
            f1 = f1^f2;
            f2 = f1 + (f1^f2);
        }
        return sum;
    }

    public String[] solutions = new String[] { "fn0" };
}
